functions/Get-DbaBackupHistory.ps1

```diff
@@ -1,449 +1,441 @@
+                    }
+                    else
+                    {
+                        Write-Message -Level SomewhatVerbose -Message "Processing as Grouped output"
+                        $GroupedResults = $results | Group-Object -Property backupsetid
+                        Write-Message -Level SomewhatVerbose -Message "$($GroupedResults.Count) result-groups found"
+                        $GroupResults = @()
+                        foreach ($group in $GroupedResults)
+                        {
+                            
+                            $FileSql = @"
+select
+	file_type as FileType,
+	logical_name as LogicalName,
+	physical_name as PhysicalName
+from
+    msdb.dbo.backupfile
+where
+    backup_set_id='$($Group.group[0].BackupSetID)'
+"@
+                            Write-Message -Level Debug -Message "FileSQL: $FileSql"
+                            
+                            $obj = New-Object sqlcollective.dbatools.Database.BackupHistory
+                            $obj.ComputerName = $server.NetName
+                            $obj.InstanceName = $server.ServiceName
+                            $obj.SqlInstance = $server.DomainInstanceName
+                            $obj.Database = $group.Group[0].Database
+                            $obj.UserName = $group.Group[0].UserName
+                            $obj.Start = ($group.Group.Start | measure-object -Minimum).Minimum
+                            $obj.End = ($group.Group.End | measure-object -Maximum).Maximum
+                            $obj.Duration = New-TimeSpan -Seconds ($group.Group.Duration | measure-object -Maximum).Maximum
+                            $obj.Path = $group.Group.Path
+                            $obj.TotalSize = ($group.group.TotalSize | measure-object -Sum).sum
+                            $obj.Type = $group.Group[0].Type
+                            $obj.BackupSetupId = $group.Group[0].BackupSetId
+                            $obj.DeviceType = $group.Group[0].DeviceType
+                            $obj.Software = $group.Group[0].Software
+                            $obj.FullName = $group.Group.Path
+                            $obj.FileList = $server.ConnectionContext.ExecuteWithResults($Filesql).Tables.Rows
+                            $obj.Position = $group.Group[0].Position
+                            $obj.FirstLsn = $group.Group[0].First_LSN
+                            $obj.DatabaseBackupLsn = $group.Group[0].database_backup_lsn
+                            $obj.CheckpointLsn = $group.Group[0].checkpoint_lsn
+                            $obj.LastLsn = $group.Group[0].Last_Lsn
+                            $obj.SoftwareVersionMajor = $group.Group[0].Software_Major_Version
+                            $GroupResults += $obj
+                        }
+                        $results = $GroupResults | Sort-Object -Property End -Descending
```

Comment:
Could this just be `$GroupResults | Sort-Object -Property End -Descending` and we skip the foreach result in results?